src/Compilers/CSharp/Portable/Symbols/Metadata/PE/NativeIntegerTypeDecoder.cs

```diff
@@ -31,11 +31,24 @@ private static TypeSymbol TransformType(TypeSymbol type, ImmutableArray<bool> tr
                 {
                     return result;
                 }
+                else
+                {
+                    return new UnsupportedMetadataTypeSymbol();
+                }
+            }
+            catch (UnsupportedSignatureContent)
+            {
+                return new UnsupportedMetadataTypeSymbol();
             }
             catch (ArgumentException)
```

Comment:
>ArgumentException [](start = 19, length = 17)

Using `ArgumentException` to manage these failure modes seems a bit dangerous, as you can't have complete confidence that it was one thrown as part of decoding in this class. It might be better to make a custom exception (private?) for this purpose.